news/php/news_article_gallery_feed.php

```diff
@@ -36,27 +36,51 @@ function create_news_article_gallery_feed($categories, $galleryStyle, $myBethel,
 //    $arrayOfNewsAndStories = get_xml($_SERVER["DOCUMENT_ROOT"] . "/_shared-content/xml/news-and-stories.xml", $categories, "inspect_news_article");
     $arrayOfNewsAndStories = sort_by_date($arrayOfNewsAndStories);
 
+    $includeStories = false;
+    $includeNews = false;
+
+    if( $newsOrStories == "Both" ) {
+        $includeStories = true;
+        $includeNews = true;
+    } elseif ( $newsOrStories == "News" ) {
+        $includeNews = true;
+    } else {
+        $includeStories = true;
+    }
+
     $threeStories = array();
-    foreach( $arrayOfNewsAndStories as $index => $newsAndStory){
+    foreach( $arrayOfNewsAndStories as $index => $newsAndStory) {
         $id = $newsAndStory['id'];
+        $addArticle = false;
 
+        // if it's already been used, skip this article
         // if its the Homepage Top Feature, skip any that aren't tagged as homepage
-        if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
+        if( in_array($id, $GLOBALS['stories-already-used']) || ($galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article']) )
             continue;
 
-        if( $newsAndStory['article-type'] == 'Story' and !in_array($id, $GLOBALS['stories-already-used'])){
+        // Check if the what the feed type is the same as the article type
+        if( $includeNews && $newsAndStory['article-type'] == 'News' ) {
+            $addArticle = true;
+        }
+        if( $includeStories && $newsAndStory['article-type'] == 'Story' ) {
```

Comment:
I suppose we could `||` lines 62 and 65 to replace 70. But either way, I like it 👍

Reply:
oh, duh 🤦‍♂️